ethdb/database.go

```diff
@@ -35,6 +35,10 @@ type KeyValueWriter interface {
 
 	// Delete removes the key from the key-value data store.
 	Delete(key []byte) error
+
+	// DeleteRange deletes all of the keys (and values) in the range [start,end)
+	// (inclusive on start, exclusive on end).
+	DeleteRange(start, end []byte) error
```

Comment:
WDYT about defining another interface `KeyValueRangeWriter`?

e.g.

```go
// KeyValueRangeWriter wraps the range write methods of a backing data store.
type KeyValueRangeWriter interface {
	// DeleteRange deletes all of the keys (and values) in the range [start,end)
	// (inclusive on start, exclusive on end).
	DeleteRange(start, end []byte) error
}

// KeyValueStore contains all the methods required to allow handling different
// key-value data stores backing the high level database.
type KeyValueStore interface {
	KeyValueReader
	KeyValueWriter
	KeyValueStater
	KeyValueRangeWriter
	Batcher
	Iteratee
	Compacter
	io.Closer
}
```

In this way, we don't need to extend all the KeyValueWriter implementations with this additional methods.

Reply:
I agree, I also didn't like how many dummy implementations I used. I called the new if `RangeDeleter` because range writer suggests that it also supports range writes which pebble actually supports but we don't need it. Anyways, it allowed me to remove most of these dummy implementations.